pandas/tests/frame/test_missing.py

```diff
@@ -388,6 +388,14 @@ def test_frame_fillna_limit(self):
         expected.values[:3] = np.nan
         tm.assert_frame_equal(result, expected)
 
+        # axis=1
+        # GH 7399
+        df.iloc[1:4, 1:4] = nan
+        expected = df.copy()
```

Comment:
needs tests with and w/o limit (and other kwargs)

Reply:
Added cases under `test_frame_fillna_axis`